From 7495a47a6ccf9361639014dbc602bc10743ef7f2 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:42:54 +0800
Subject: [PATCH 09/10] GSI Android 17 compatibility:
 0010-Samsung-SMSC-When-we-receive-those-weird-SMSC-actual.patch

Change-Id: Icd9166727db4f7998cbf48a5db53a60468afd2fc
---
 .../internal/telephony/MessagingResponse.java       | 13 +++++++++++--
 .../android/internal/telephony/RadioResponse.java   | 13 +++++++++++--
 2 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/MessagingResponse.java b/src/java/com/android/internal/telephony/MessagingResponse.java
index 3ddf676..2e62599 100644
--- a/src/java/com/android/internal/telephony/MessagingResponse.java
+++ b/src/java/com/android/internal/telephony/MessagingResponse.java
@@ -170,8 +170,17 @@ public class MessagingResponse extends IRadioMessagingResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
-        if(smsc.contains("\"") || smsc.contains(","))
-            smsc = "";
+        if(smsc.contains("\"") || smsc.contains(",")) {
+            android.util.Log.e("PHH", "Got weird SMSC: " + smsc);
+            try {
+                String[] a = smsc.split("\"");
+                smsc = a[1];
+            } catch(Throwable t) {
+                android.util.Log.e("PHH", "Failed parsing weird smsc", t);
+                smsc = "";
+            }
+            android.util.Log.e("PHH", "Patched smsc " + smsc);
+        }
         RadioResponse.responseString(HAL_SERVICE_MESSAGING, mRil, responseInfo, smsc);
     }
 
diff --git a/src/java/com/android/internal/telephony/RadioResponse.java b/src/java/com/android/internal/telephony/RadioResponse.java
index 5663dc8..e470e68 100644
--- a/src/java/com/android/internal/telephony/RadioResponse.java
+++ b/src/java/com/android/internal/telephony/RadioResponse.java
@@ -1237,8 +1237,17 @@ public class RadioResponse extends IRadioResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
-        if(smsc.contains("\"") || smsc.contains(","))
-            smsc = "";
+        if(smsc.contains("\"") || smsc.contains(",")) {
+            android.util.Log.e("PHH", "Got weird SMSC: " + smsc);
+            try {
+                String[] a = smsc.split("\"");
+                smsc = a[1];
+            } catch(Throwable t) {
+                android.util.Log.e("PHH", "Failed parsing weird smsc", t);
+                smsc = "";
+            }
+            android.util.Log.e("PHH", "Patched smsc " + smsc);
+        }
         responseString(responseInfo, smsc);
     }
 
